05_Datathon/diabetes_prediction_example.Rmd

```diff
@@ -0,0 +1,336 @@
+```
+
+Next we will encode our categorical variables. Encoding is the process of reshaping and binarizing categorical data to better suit machine learning models.
+
+```{r}
+
+# Convert ethnicity and diabetes_mellitus to factor
+gossis$ethnicity <- as.factor(gossis$ethnicity)
+gossis$diabetes_mellitus <- as.factor(gossis$diabetes_mellitus)
+
+# Encode gender variable: male = 1, female = 0
+gossis$gender <- ifelse(gossis$gender == "M",1,0)
+```
+
+
+## 3. Create the training and testing sets
+
+We will only include four predictors in our model: `age`, `bmi`, `gender`, and `glucose apache`. Feel free to include more or choose entirely different predictors in your model!
+
+```{r training_test, echo=TRUE}
+library(caret)
+library(dplyr)
+
+# Set the random number seed for reproducibility
+set.seed(1)
+
+# CHOOSE VARIABLE YOU WANT TO PREDICT HERE!
+reduced_data$target_variable <- as.factor(reduced_data$diabetes_mellitus)
+
+# Create data partition using the target variable
+train_index <- createDataPartition(reduced_data$target_variable, times = 1, p = 0.8, list = FALSE)
+
+# Split data into train and test sets, select columns that will be used in model
+train_set <- reduced_data[train_index, ] %>% select(., age, bmi, gender, ethnicity,glucose_apache,icu_type, target_variable)
+head(train_set)
+test_set <- reduced_data[-train_index, ] %>% select(., age, bmi, gender, ethnicity,glucose_apache ,icu_type, target_variable)
+head(test_set)
+```
+
+## 4. Calculate summary statistics ("Table 1")
+
+- Most studies include summary statistics as Table 1.
+- `library(tableone)` makes it easy to create these summary statistics
+- Developed by Dr Kazuki Yoshida, while a Ph.D. student at Harvard University.
+- [TableOne documentation](https://cran.r-project.org/web/packages/tableone/tableone.pdf)
+
+```{r tableone, echo=TRUE}
+library(tableone)
+
+allvars = c("bmi", "age", "gender", "ethnicity","glucose_apache","icu_type")
+catvars = c("ethnicity","gender", "icu_type")
+
+table_gossis <- CreateTableOne(vars = allvars, data = train_set, 
+                factorVars = catvars,
+                strata="target_variable")
+
+kableone(table_gossis)
+```
+
+## 5. Model Construction
+
+### 5.1 Train a random forest classifier
+
+- We will use the `randomForest` library to build a random forest model.
+- See the [randomForest documentation on CRAN](https://cran.r-project.org/web/packages/randomForest/randomForest.pdf).
+
+```{r train_model, echo=TRUE}
+# Define forest tuning parameters
+
+# 5-fold cross validation
+control <- trainControl(method='repeatedcv', 
+                        number=5, 
+                        repeats=3)
+
+# Number of variables tried at each split
+mtry <- sqrt(ncol(train_set))
+
+# Grid search is a linear search through a vector of mtry values
+tunegrid <- expand.grid(.mtry=mtry)
+
+# Create classification forest using age, bmi, and gender
+forest <- randomForest(target_variable ~ age + bmi + gender + glucose_apache, 
```

Comment:
I think using caret here is worthwhile because it's generic to be able to have access to many models, and requires minimal changes:

```
train(target_variable ~ age + bmi + gender + glucose_apache, 
               data = train_set,
               metric = "Accuracy",
               tuneGrid = tunegrid,
               trControl = control,method="rf")